### PR TITLE
Add testing for edge cases of name conversions in grpc

### PR DESCRIPTION
**Pull request by peter-edge**

### CODE REVIEW

File: transport/x/grpc/util_test.go [hunk 1]
```diff
@@ -66,3 +67,21 @@ func TestProcedureNameFunctions(t *testing.T) {
...
+
+func TestProcedureNameEmpty(t *testing.T) {
+	_, _, err := procedureNameToServiceNameMethodName("")
+	require.Error(t, err)
+	_, err = procedureNameToFullMethod("")
+	require.Error(t, err)
+}
+
+func TestToFullMethodEmptyServiceName(t *testing.T) {
+	require.Equal(t, "/__default__/foo", toFullMethod("", "foo"))
```
Comment (kriskowal) [inline transport/x/grpc/util_test.go]: A reminder that I remain not a fan of this `__default__/name` scheme, but not a blocker. I would like to hear about alternatives you’re discarding and why. Do we have an opportunity to make a better decision here if we pursue @willhug’s work on bypassing gRPC’s encoding and route table equivalents?

Comment (kriskowal) [inline transport/x/grpc/util_test.go]: (Ideally the gRPC method would be equivalent to YARPC procedure, and not force Protobuf opinions about procedure name structure on other encodings. But, I’m sure we can’t have nice things.)

Comment (peter-edge) [inline transport/x/grpc/util_test.go]: I'm not a fan either, I think we should discuss it when I get into SF for sure
